plinth/frontpage.py

```diff
@@ -27,18 +27,31 @@ def get_shortcuts():
     return sorted(shortcuts.values(), key=lambda item: item['label'])
 
 
-def add_shortcut(id, label, url, icon, details=None):
+def add_shortcut(id, label, url, icon, details=None,login_required=True):
```

Comment:
Space is missing after the comma. For coding standards see: https://www.python.org/dev/peps/pep-0008/ and https://www.python.org/dev/peps/pep-0257/.  You should integrate flake8 tool into your editor so that such errors are automatically reported. For Emacs, having enabling flycheck-mode with flake8 tool installed in the system will do the trick.